packages/@ember/-internals/glimmer/lib/glimmer-tracking-docs.ts

```diff
@@ -196,43 +196,46 @@
     }
   ```
 
-  Now the sortedGuests getter will be cached based on autotracking.
+  Now the `sortedGuests` getter will be cached based on autotracking.
   It will only rerun and create a new sorted array when the guests tracked
   property is updated.
 
+
   ### Tradeoffs
 
   Overuse is discouraged.
 
   In general, you should avoid using `@cached` unless you have confirmed that
-  the getter you are decorating is computationally expensive. `@cached` adds
-  a small amount of overhead to the getter, making it more expensive.
-  While this overhead is small, if `@cached` is overused it can add up to a
-  large impact overall in your app. Many getters and tracked properties
-  are only accessed once, rendered, and then never rerendered, so adding
-  `@cached` when it is unnecessary can negatively impact performance.
+  the getter you are decorating is computationally expensive, since `@cached`
+  adds a small amount of overhead to the getter.
+  While the individual costs are small, a systematic use of the `@cached`
+  decorator can add up to a large impact overall in your app.
+  Many getters and tracked properties are only accessed once during rendering,
+  and then never rerendered, so adding `@cached` when unnecessary can
+  negatively impact performance.
 
   Also, `@cached` may rerun even if the values themselves have not changed,
-  since tracked properties will always invalidate even if their underlying
-  value did not change.
-  For example updating an integer value from `5` to an other `5`.
+  since tracked properties will always invalidate.
+  For example updating an integer value from `5` to an other `5` will trigger
+  a rerun of the cached properties building from this integer.
 
   Avoiding a cache invalidation in this case is not something that can
   be achieved on the `@cached` decorator itself, but rather when updating
-  the underlying values, by applying a diff checking mecanism:
+  the underlying tracked values, by applying some diff checking mecanisms:
 
   ```javascript
-  if (newValue !== this.trackedProp) {
-    this.trackedProp = newValue;
+  if (nextValue !== this.trackedProp) {
+    this.trackedProp = nextValue;
```

Comment:
a caveat may need to be mentioned here -- this can cause the infinite re-validation assertion message if this code exists in a getter (you can't modify a "tracked prop used during render" during render)

Reply:
Done !
It seemed like a weird move to set properties from a getter but considering it would be from a cached one its not impossible to build some spaghetti code/logic on top of it.
The end result would scream to be put into a resource, so I see where you got the idea for the caveat from 😉.
Once resource/use becomes available, I will be amending this file to reference the new solution for the more complex needs.